### ISSUE TITLE
Custom search engines not functioning from Vomnibar.activateInNewTab

### ISSUE THREAD

**Issue by skoogit** [philc/vimium#1461]
Custom search engines does not work at all when using the Vomnibar.activateInNewTab, the search is simply performed in the default search engine.

E.g. "w searchterm" performs a (in my case) DuckDuckGo search for "w searchterm" rather than searching wikipedia for "searchterm".

**Comment by smblott-github** [comment]
@skoogit.  Thanks for the report.  I see odd things in 1.49, but they seem to have been fixed in `master`.

Odd things in 1.49...
- For the very first search in a new tab, `w bbc` highlights the search-engine suggestion, and uses it on `Enter`.  Seems to be the same for `o` and `O`.
- Subsequent searches (with either `o` or `O`) *do not* highlight the search-engine suggestion, and the user has to highlight it manually (with `Tab`, say) and then hit `Enter`.

Is that what you're seeing?

As I said, this seems to have been fixed in `master`.

**Comment by skoogit** [comment]
It works as you say for the first `O` search on each tab, but any subsequent `O` searches from that tab does not. Your right about the highlightning too. This problem prevails until the tab which have already performed a `O` search changes url.

However, it's not the same case for both `o` and `O` for me. `o` searches works perfect for me in all cases (ignoring the >1 word issue) except one and does not in any way hickup the `O` searches.

The only case where a `o` search doesn't work for me is for the first search after performing two subsequent `O` searches. However following up with another `o` search works, and following that up with another `O` search works too:
Tab1 `O` = OK
Tab1 `O` = ERROR
Tab1 `o` = ERROR
Tab1 `o` = OK
Tab1 'O' = OK

**Comment by smblott-github** [comment]
This seems to be fixed in `master`, and #1462 will probably soon be merged (which fixes another problem I uncovered while looking into your issue).  Thanks for the report and the update.

**Comment by saheel1115** [comment]
@smblott-github Will we get an update notification after the merge? If not, how do we get this fix in? I am also facing this problem and would like to update Vimium after the merge.

**Comment by skoogit** [comment]
Closed accidently, maybe saheel's waiting for an answer first. (as stated, not really at home with github routines yet, so uncertain)

**Comment by smblott-github** [comment]
@saheel1115.  Being merged into `master` is just the first step.  If you use the Chrome Store release... then I'm afraid you'll have to wait for the next release.  I'm not sure when that will be.  @philc's the mastermind there.  I'm going to guess a couple of months.  We're working on some new features (e.g. visual mode), and they may require a little time to get the UX right, and shake out any bugs.

**Comment by saheel1115** [comment]
@smblott-github Okay, so I pulled `master` and installed Vimium in Developer Mode using the instructions on the Contributing.md page. Now, the custom search engines don't work at all! :( (Earlier, in the Chrome Web Store release, the 1st custom search engine would work but not the others.) Other functionalities seem to work fine.

Note: Only 1 instance of Vimium is enabled and I have tried rebooting the browser.

**Comment by smblott-github** [comment]
@saheel1115. Could you post your custom search engine configuration, please?

**Comment by saheel1115** [comment]
My bad. The thing is, in the `master` version, until you enter a term to search it does not say that it's going to search for that particular term. For example, after pressing "w ", it gives no indication that it is going to search wikipedia. Only after I enter the first letter of the search term ('s' if I want to search for 'something') does it show (and highlight it as well) that it is going to search wikipedia. This behavior, which is different from the Web Store release, tipped me off and hence I reported initially that custom search engines were not working at all.

Knowing the above 'problem', I properly tested and all the search engines in my list seem to work now. Pics attached.

![vimium_success](https://cloud.githubusercontent.com/assets/834236/6086192/b345001a-adf4-11e4-84fd-3c2cf3990b75.png)
![vimium_fail](https://cloud.githubusercontent.com/assets/834236/6086191/b2f92d3e-adf4-11e4-842e-d2cb2c6c4d09.png)

**Comment by smblott-github** [comment]
Closing, now.  Looks like all of the issues have been addressed.
